### PR TITLE
fix(query): do not visit dehydrated injectors.

### PR DESCRIPTION
**Pull request by rkirov**

### CODE REVIEW

File: modules/angular2/test/core/compiler/query_integration_spec.ts [hunk 1]
```diff
@@ -103,6 +103,26 @@ export function main() {
                });
          }));
 
+      it('should be cleanly destroyed',
```
Comment (IgorMinar) [inline modules/angular2/test/core/compiler/query_integration_spec.ts]: can you make this more descriptive please?

maybe: 'should be cleanly destroyed when children cross view boundaries'

**Comment by IgorMinar** [comment]
otherwise lgtm

File: modules/angular2/test/core/compiler/query_integration_spec.ts (hunk 1 above)
Comment (rkirov) [inline modules/angular2/test/core/compiler/query_integration_spec.ts]: done

**Comment by IgorMinar** [comment]
Can you fix the test failures before Monday noon please?

**Comment by rkirov** [comment]
Should be good now, had to add one more line change. PTAL.

**Comment by IgorMinar** [comment]
lgtm

**Pull request closed without merging**